uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/Service/LeaveManagerTest.php

```diff
@@ -25,109 +29,109 @@ public function tearDown() {
+    $this->assertFalse($this->leaveManagerService->isContactManagedBy($this->staffMember['id'], $this->manager['id']));
 
     $today = new DateTime('today');
     $tomorrow = new DateTime('tomorrow');
-    $this->setContactAsLeaveApproverOf($manager, $staffMember, $today->format('Y-m-d'), $tomorrow->format('Y-m-d'));
+    $this->setContactAsLeaveApproverOf($this->manager, $this->staffMember, $today->format('Y-m-d'), $tomorrow->format('Y-m-d'));
 
-    $this->assertTrue($this->leaveManagerService->isContactManagedBy($staffMember['id'], $manager['id']));
+    $this->assertTrue($this->leaveManagerService->isContactManagedBy($this->staffMember['id'], $this->manager['id']));
   }
 
   public function testIsContactManagedByWhenTheresNoActiveRelationshipForTheCurrentDate() {
-    $manager = ContactFabricator::fabricate();
-    $staffMember = ContactFabricator::fabricate();
-
-    $this->assertFalse($this->leaveManagerService->isContactManagedBy($staffMember['id'], $manager['id']));
+    $this->assertFalse($this->leaveManagerService->isContactManagedBy($this->staffMember['id'], $this->manager['id']));
 
     // Set a relationship in the past
     $startDate = new DateTime('-10 days');
     $endDate = new DateTime('-1 day');
-    $this->setContactAsLeaveApproverOf($manager, $staffMember, $startDate->format('Y-m-d'), $endDate->format('Y-m-d'));
+    $this->setContactAsLeaveApproverOf($this->manager, $this->staffMember, $startDate->format('Y-m-d'), $endDate->format('Y-m-d'));
 
-    $this->assertFalse($this->leaveManagerService->isContactManagedBy($staffMember['id'], $manager['id']));
+    $this->assertFalse($this->leaveManagerService->isContactManagedBy($this->staffMember['id'], $this->manager['id']));
 
     // Set a relationship in the future
     $startDate = new DateTime('+1 day');
-    $this->setContactAsLeaveApproverOf($manager, $staffMember, $startDate->format('Y-m-d'));
+    $this->setContactAsLeaveApproverOf($this->manager, $this->staffMember, $startDate->format('Y-m-d'));
 
-    $this->assertFalse($this->leaveManagerService->isContactManagedBy($staffMember['id'], $manager['id']));
+    $this->assertFalse($this->leaveManagerService->isContactManagedBy($this->staffMember['id'], $this->manager['id']));
   }
 
   public function testIsContactManagedByWhenTheCurrentRelationshipIsNotActive() {
-    $manager = ContactFabricator::fabricate();
-    $staffMember = ContactFabricator::fabricate();
+    $this->assertFalse($this->leaveManagerService->isContactManagedBy($this->staffMember['id'], $this->manager['id']));
+
+    $this->setContactAsLeaveApproverOf($this->manager, $this->staffMember, null, null, false);
+
+    $this->assertFalse($this->leaveManagerService->isContactManagedBy($this->staffMember['id'], $this->manager['id']));
+  }
 
-    $this->assertFalse($this->leaveManagerService->isContactManagedBy($staffMember['id'], $manager['id']));
+  public function testIsContactManagedByWhenCurrentUserHasMoreThanOneRelationshipWithTheEmployee() {
```

Comment:
The test name is a bit misleading. The test actually has nothing to do with the current user. The `isContactManagedBy` takes two contact parameters irrespective of if any of the contact is the current/loggen in user or not.
Also the phrase `HasMoreThanOneRelationshipWithTheEmployee` makes one think its possible for the manager to have more than one relationship with the employee at a given time which seems not possible?